modules/memfault/Kconfig

```diff
@@ -0,0 +1,132 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+if MEMFAULT
+
+config MEMFAULT_NCS_USE_DEFAULT_METRICS
+	bool
+	select MEMFAULT_METRICS_EXTRA_DEFS_FILE
+	default y if (MEMFAULT_NCS_STACK_METRICS || MEMFAULT_NCS_LTE_METRICS)
+
+config MEMFAULT_NCS_PROJECT_KEY
+	string "Memfault API key"
+
+config MEMFAULT_NCS_DEVICE_ID
+	string "Memfault device ID"
+
+config MEMFAULT_NCS_DEVICE_SERIAL_USE_IMEI
+	bool "Use IMEI as Memfault device serial"
+	depends on AT_CMD
+	default y if BOARD_NRF9160DK_NRF9160NS || BOARD_THINGY91_NRF9160NS
+
+config MEMFAULT_NCS_HW_VERSION
+	string "Hardware version"
+	default "nrf9160dk" if BOARD_NRF9160DK_NRF9160NS
+	default "thingy91" if BOARD_THINGY91_NRF9160NS
+
+config MEMFAULT_NCS_FW_TYPE
+	string "Firmware type"
+	default "nrf91ns-fw" if BOARD_NRF9160DK_NRF9160NS || BOARD_THINGY91_NRF9160NS
+	help
+	  Firmware type running on the board.
+
+choice
+	prompt "Firmware version generation method"
+	default MEMFAULT_NCS_FW_VERSION_AUTO
+
+config MEMFAULT_NCS_FW_VERSION_STATIC
+	bool "Static firmware version"
+	help
+	  Use statically configured firmware version, as opposed to an
+	  dynamically generated one.
+
+config MEMFAULT_NCS_FW_VERSION_AUTO
+	bool "Dynamic firmware version"
+	help
+	  Use dyanimically generated firmware version. The version is
+	  re-generated on every compiler run.
+endchoice
+
+config MEMFAULT_NCS_FW_VERSION_PREFIX
+	string "Firmware version prefix"
+	depends on MEMFAULT_NCS_FW_VERSION_AUTO
+	default "0.0.1+"
+	help
+	  Prefix to use in front of automatically generated build version string.
+	  If the build version string is "a1b2c3d", the full firmware version
+	  will become "<prefix>a1b2c3d".
+
+config MEMFAULT_NCS_INIT_PRIORITY
+	int "memfault initialization priority"
+	range AT_CMD_INIT_PRIORITY 99 if MEMFAULT_NCS_DEVICE_SERIAL_USE_IMEI
+	default 90
+	help
+	  If IMEI is used as device ID, the initialization priority must be
+	  greater than that of the AT command module.
+
+config MEMFAULT_NCS_PROVISION_CERTIFICATES
+	bool "Provision certificates on initalization"
+	depends on  MEMFAULT_ROOT_CERT_STORAGE_NRF9160_MODEM
+	default y
+	help
+	  Provision required TLS root certificates for connecting to Memfault's
+	  endpoints. If this option is disabled, the user is responsible for
+	  calling the appropriate Memfault APIs for provisioning the
+	  certificates before attempting to connect to Memfault servers.
+
+config MEMFAULT_NCS_STACK_METRICS
+	bool "Collect stack metrics"
+	select THREAD_ANALYZER
+	select THREAD_NAME
+	default y
+	help
+	  Collect metrics for unused stack space for selected stacks.
+	  Currently the following stacks' unused space is monitored:
+		- at_cmd_socket_thread, used by the AT command library
+		- connection_poll_thread, used by the cloud libraries for
+		  nRF Cloud, AWS IoT and Azure IoT Hub
+
+config MEMFAULT_NCS_STACK_CHECK_INTERVAL
```

Comment:
Could add `_SECONDS` or `_SEC` as suffix to make it clear what resolution the interval is in.